### PR TITLE
Move getAdjustedTokenCountFromModel from CLI to core

### PR DESCRIPTION
**Pull request by RomneyDa**
## Summary

This PR moves the `getAdjustedTokenCountFromModel` function from the CLI tokenizer to `core/llm/countTokens.ts` and integrates it with core's token counting logic.

## Changes

- **Moved function**: Extracted `getAdjustedTokenCountFromModel` from CLI to core
- **Integration**: Updated core's `countTokens()` to apply model-specific multipliers
- **Tests**: Added comprehensive test suite for the function in core
- **CLI Update**: Updated CLI tokenizer to import from core instead of duplicating code
- **Cleanup**: Removed duplicate constants and implementation from CLI

## Why?

- **DRY**: Eliminates code duplication between CLI and core
- **Consistency**: Ensures token counting uses the same multipliers everywhere
- **Maintainability**: Single source of truth for token adjustment logic
- **Better coverage**: The llama tokenizer (~= gpt tokenizer) now applies safety multipliers in core as well

## Model Multipliers

The function applies these multipliers based on empirical data:
- **Claude**: 1.23x (Anthropic tokenizer produces ~23% more tokens)
- **Gemini**: 1.18x (Gemini tokenizer produces ~18% more tokens)
- **Mistral family**: 1.26x (includes mistral, mixtral, devstral, codestral)

## Testing

Added unit tests covering:
- Base cases (no multiplier)
- All model-specific multipliers
- Edge cases (zero tokens, undefined model names)
- Case-insensitivity

## Breaking Changes

None - this is backward compatible. The function signature changed slightly (takes model name string instead of ModelConfig) but all call sites have been updated.

---

**Continue Tasks:** ▶️ 2 queued · ✅ 2 no changes — [View all](https://hub.continue.dev/inbox?pr=https%3A%2F%2Fgithub.com%2Fcontinuedev%2Fcontinue%2Fpull%2F9968&utm_source=github_pr&utm_medium=pr_body&utm_campaign=continue_tasks)

---
## Summary by cubic
Moves getAdjustedTokenCountFromModel to core and applies model-specific multipliers inside core token counting. This unifies token estimates across core and CLI.

- **Refactors**
  - Centralized getAdjustedTokenCountFromModel in core/llm/getAdjustedTokenCount.ts and integrated it into countTokens().
  - Applied model multipliers: Claude 1.23x, Gemini 1.18x, Mistral family 1.26x.
  - CLI now imports from core; removed duplicate logic and constants.
  - Added unit tests for multipliers and edge cases. No breaking changes.

<sup>Written for commit 318cb66c5e412fbea12d5d8fc700f43cea291b70. Summary will update on new commits.</sup>

### CODE REVIEW

File: core/llm/countTokens.ts [hunk 1]
```diff
@@ -124,8 +161,9 @@ function countTokens(
       );
     }, 0);
   } else {
-    return encoding.encode(content ?? "", "all", []).length;
+    baseTokens = encoding.encode(content ?? "", "all", []).length;
   }
+  return getAdjustedTokenCountFromModel(baseTokens, modelName);
```
[the remark this answers is not in the document]
Comment (RomneyDa) [reply core/llm/countTokens.ts:131]: ✅ Fixed! You're absolutely right - `countTokensAsync` was not applying the model-specific token multipliers that `countTokens` uses.

**The issue:** 
- `countTokens` applied multipliers for Claude (1.23x), Gemini (1.18x), and Mistral (1.26x) models
- `countTokensAsync` did NOT apply these multipliers
- This caused async token counts to be ~18-26% lower for these models
- Affected limit checks like `readFileLimit` which could incorrectly accept files that are too large

**The fix:**
Modified `countTokensAsync` to call `getAdjustedTokenCountFromModel` just like `countTokens` does. Both functions now return identical token counts for the same model and content.

**Testing:**
Added comprehensive tests to verify:
- Multipliers are correctly applied for Claude, Gemini, and Mistral models
- `countTokensAsync` and `countTokens` return identical results
- Base models (like GPT-4) are unaffected

Thanks for catching this inconsistency!

File: extensions/cli/src/util/tokenizer.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { ModelConfig } from "@continuedev/config-yaml";
 import type { ChatHistoryItem } from "core/index.js";
+import { getAdjustedTokenCountFromModel } from "core/llm/countTokens.js";
```
[the remark this answers is not in the document]
Comment (RomneyDa) [reply extensions/cli/src/util/tokenizer.ts]: Great catch! You're absolutely right - importing from `core/llm/countTokens.js` was pulling in heavy tokenizer dependencies (js-tiktoken, llamaTokenizer) at module load time for what is essentially just a simple multiplier function.

**Fix:**
I've extracted `getAdjustedTokenCountFromModel` to a new lightweight utility file `core/llm/utils/tokenAdjustment.ts` that contains only the multiplier logic with no tokenizer dependencies. The CLI now imports from this utility instead.

**Changes:**
- Created `core/llm/utils/tokenAdjustment.ts` - pure multiplier logic, no heavy deps
- Updated CLI to import from the lightweight utility
- Updated `core/llm/countTokens.ts` to import and re-export from the utility (maintains backward compatibility)
- Added comprehensive tests for the token adjustment logic

This prevents the CLI from loading ~700KB+ of tokenizer code at startup when it only needs a simple model-specific multiplier. The countTokens module still works exactly the same way, but now shares the lightweight multiplier utility with the CLI.

**Review by sestinj — APPROVED** [review]
